### PR TITLE
Update cdflib to 0.3.20 and update version check for future releases of cdflib

### PR DESCRIPTION
**Pull request by nabobalis**
Should fix the CI.

### CODE REVIEW

**Comment by nabobalis** [comment]
There are still issues but I think they are now upstream in cdflib. At least one of them might be.

**Comment by dstansby** [comment]
I've just released v1.0.0 of cdflib, so have re-started the devdeps build to see if it works.

**Comment by dstansby** [comment]
Hmm, I think `cdflib` might have issues with the un-released numpy 1.25, which is causing the errors here.

File: setup.cfg [hunk 1]
```diff
@@ -105,6 +105,8 @@ tests =
   pytest-mpl>=0.12 # First version to support our figure tests
   pytest-xdist>=2.0
   pytest>=6.0
+  # cdflib needs this for now
+  xarray
```
Comment (dstansby) [inline setup.cfg:109]:
```suggestion
```

File: setup.cfg [hunk 2]
```diff
@@ -74,7 +74,7 @@ net =
   tqdm>=4.32.1
   zeep>=3.4.0
 timeseries =
-  cdflib>=0.3.19,!=0.4.0
+  cdflib>=0.3.20,!=0.4.0
```
Comment (dstansby) [inline setup.cfg:77]:
```suggestion
  cdflib>=0.3.20,!=0.4.0,!=1.0.0
```

File: sunpy/io/_cdf.py [hunk 3]
```diff
@@ -44,8 +44,7 @@ def read_cdf(fname):
     # Extract the time varying variables
     cdf_info = cdf.cdf_info()
     meta = cdf.globalattsget()
-    cdflib_version = Version(cdflib.__version__)
-    if cdflib_version > Version('0.4.9'):
+    if hasattr(cdflib, "__version__") and Version(cdflib.__version__) > Version("0.4.9"):
```
Comment (dstansby) [inline sunpy/io/_cdf.py:47]:
```suggestion
    if hasattr(cdflib, "__version__") and Version(cdflib.__version__) < Version("1.0.0"):
```

File: sunpy/io/_cdf.py [hunk 4]
```diff
@@ -79,7 +75,7 @@ def read_cdf(fname):
                 continue
 
             # Get data
-            if cdflib_version > Version('0.4.9'):
+            if hasattr(cdflib, "__version__") and Version(cdflib.__version__) > Version("0.4.9"):
```
Comment (dstansby) [inline sunpy/io/_cdf.py:78]:
```suggestion
            if hasattr(cdflib, "__version__") and Version(cdflib.__version__) < Version("1.0.0"):
```

File: sunpy/io/_cdf.py [hunk 5]
```diff
@@ -79,7 +75,7 @@ def read_cdf(fname):
                 continue
 
             # Get data
-            if cdflib_version > Version('0.4.9'):
+            if hasattr(cdflib, "__version__") and Version(cdflib.__version__) < Version("1.0.0"):
```
Comment (dstansby) [inline sunpy/io/_cdf.py:78]:
```suggestion
            if hasattr(cdflib, "__version__") and Version(cdflib.__version__) >= Version("1.0.0"):
```

File: sunpy/io/_cdf.py [hunk 6]
```diff
@@ -44,8 +44,7 @@ def read_cdf(fname):
     # Extract the time varying variables
     cdf_info = cdf.cdf_info()
     meta = cdf.globalattsget()
-    cdflib_version = Version(cdflib.__version__)
-    if cdflib_version > Version('0.4.9'):
+    if hasattr(cdflib, "__version__") and Version(cdflib.__version__) < Version("1.0.0"):
```
Comment (dstansby) [inline sunpy/io/_cdf.py:47]:
```suggestion
    if hasattr(cdflib, "__version__") and Version(cdflib.__version__) >= Version("1.0.0"):
```

**Comment by dstansby** [comment]
I guess I'll have to set up a devdeps build for cdflib, but for now I think worth merging this to get the cdflib version checking a bit more accurate for sunpy 5.0

**Review by dstansby — APPROVED** [review]

**Pull request merged**